libsolidity/analysis/TypeChecker.cpp

```diff
@@ -487,7 +489,10 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 
 bool TypeChecker::visit(VariableDeclaration const& _variable)
 {
-	if (m_scope->contractKind() == ContractDefinition::ContractKind::Interface)
+	if (
+		m_scope->contractKind() == ContractDefinition::ContractKind::Interface
+		&& !_variable.isCallableParameter()
```

Comment:
Actually can you add a comment here to explain what is this for? (As a good practice, because the `isCallableParameter` doesn't seem very meaningful in this context)